File: Backend/PrestationServices/src/main/java/tn/esprit/Entities/Partenaire.java
package tn.esprit.Entities;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.Date;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Partenaire {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private int idparte;
    @Temporal(TemporalType.DATE)
    private Date datepartenariat ;
    @Enumerated(EnumType.STRING)
    private Statut statut ;

    @ManyToOne
    private Entreprise entreprise1;
    @ManyToOne
    private Entreprise entreprise2;


}

File: Backend/Frontend/src/app/models/Partenaire/partenaire.ts
import { Entreprise } from "../Entreprise/entreprise";

export class Partenaire {
    idParte!: number;
    datePartenariat!: Date;
    statut!: Statut;
    entreprise1!: Entreprise;
    entreprise2!: Entreprise;
  }
  export enum Statut {
    EN_COURS = 'ENCOURS',
    TERMINEE = 'TERMINEE',
  }